client/QuickStart/API_Util.cs

```diff
@@ -0,0 +1,410 @@
+
+namespace QuickStart
+{
+    /// <summary>
+    /// ConfigJson object - represents a cluster and DataBase connection configuration file.
+    /// </summary>
+    public class ConfigJson
+    {
+        /// Flag to indicate whether to use an existing table, or to create a new one. 
+        public bool UseExistingTable { get; set; }
+
+        /// DB to work on from the given URI.
+        public string DatabaseName { get; set; }
+
+        /// Table to work on from the given DB.
+        public string TableName { get; set; }
+
+        /// Table to work with from the given Table.
+        public string TableSchema { get; set; }
+
+        /// Cluster to connect to and query from.
+        public string KustoUri { get; set; }
+
+        /// Ingestion cluster to connect and ingest to. will usually be the same as the KustoUri, but starting with "ingest-"...
+        public string IngestUri { get; set; }
+
+        /// Certificate Path when using AppCertificate authentication mode.
+        public string CertificatePath { get; set; }
+
+        /// Certificate Password when using AppCertificate authentication mode.
+        public string CertificatePassword { get; set; }
+
+        /// Application Id when using AppCertificate authentication mode.
+        public string ApplicationId { get; set; }
+
+        /// Tenant Id when using AppCertificate authentication mode.
+        public string TenantId { get; set; }
+
+        /// Data sources list to ingest from.
+        public List<Dictionary<string, string>> Data { get; set; }
+
+        /// Flag to indicate whether to Alter-merge the table (query).
+        public bool AlterTable { get; set; }
+
+        /// Flag to indicate whether to query the starting data (query).
+        public bool QueryData { get; set; }
+
+        /// Flag to indicate whether ingest data based on data sources.
+        public bool IngestData { get; set; }
+
+        /// Recommended default: UserPrompt
+        /// Some of the auth modes require additional environment variables to be set in order to work (see usage in generate_connection_string function).
+        /// Managed Identity Authentication only works when running as an Azure service (webapp, function, etc.)
+        /// Options: (UserPrompt|ManagedIdentity|AppKey|AppCertificate)
+        public string AuthenticationMode { get; set; }
+
+        /// Recommended default: True
+        /// Toggle to False to execute this script "unattended"
+        public bool WaitForUser { get; set; }
+
+        /// Sleep time to allow for queued ingestion to complete.
+        public int WaitForIngestSeconds { get; set; }
+
+        /// Optional - Customized ingestion batching policy
+        public string BatchingPolicy { get; set; }
+    }
+
+    /// <summary>
+    /// Util static class - Handels the communication with the API, and provides generic and simple "plug-n-play" functions to use in different programs.
+    /// </summary>
+    public static class Util
```

Comment:
Name change - done.
This is what we discussed previously - create one class to do the proper SampleApp actions, and one that handles all the API calls and envelops it for the user. i can try and split it further (maybe into two utils - one for execution and one for ingestion), but ill admit i think it will be a bit redundant, as it will start to be a lot like the SDK itself.

Reply:
Looks good